hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisSqmFunctionDescriptors.java

```diff
@@ -7,22 +7,78 @@
 
 package org.hibernate.spatial.dialect.postgis;
 
-import java.util.Arrays;
+import java.util.List;
 
 import org.hibernate.boot.model.FunctionContributions;
 import org.hibernate.query.sqm.function.NamedSqmFunctionDescriptor;
+import org.hibernate.query.sqm.produce.function.ArgumentsValidator;
 import org.hibernate.query.sqm.produce.function.FunctionReturnTypeResolver;
-import org.hibernate.query.sqm.produce.function.StandardArgumentsValidators;
 import org.hibernate.query.sqm.produce.function.StandardFunctionReturnTypeResolvers;
 import org.hibernate.spatial.BaseSqmFunctionDescriptors;
-import org.hibernate.spatial.CommonSpatialFunction;
+import org.hibernate.spatial.FunctionKey;
+import org.hibernate.sql.ast.SqlAstNodeRenderingMode;
+import org.hibernate.sql.ast.SqlAstTranslator;
+import org.hibernate.sql.ast.spi.SqlAppender;
+import org.hibernate.sql.ast.tree.SqlAstNode;
+import org.hibernate.sql.ast.tree.expression.Expression;
 import org.hibernate.type.BasicTypeRegistry;
+import org.hibernate.type.StandardBasicTypes;
+
+import static org.hibernate.query.sqm.produce.function.StandardArgumentsValidators.exactly;
 
 public class PostgisSqmFunctionDescriptors extends BaseSqmFunctionDescriptors {
 
+	private final BasicTypeRegistry typeRegistry;
+
 	public PostgisSqmFunctionDescriptors(FunctionContributions functionContributions) {
 		super( functionContributions );
+		typeRegistry = functionContributions.getTypeConfiguration().getBasicTypeRegistry();
+		addOperator("distance_2d", "<->");
+		addOperator("distance_2d_bbox", "<#>");
+		addOperator("distance_cpa", "|=|");
+		addOperator( "distance_centroid_nd", "<<->>" );
+		// <<#>> operator is apparently no longer supported?
+		//addOperator( "distance_nd_bbox", "<<#>>" );
+	}
+
+	protected void addOperator(String name, String operator) {
+		map.put(
+				FunctionKey.apply( name ),
+				new PostgisOperator(
+						name,
+						operator,
+						exactly( 2 ),
```

Comment:
Would be awesome if you could add a custom `ArgumentsValidator` that validates the types of the operands.

Reply:
I didn't opt for a custom `ArgumentsValidator` but extended the approach I saw in e.g. the avg function implementation. Could you have a look?